pkg/check/pss/pss.go

```diff
@@ -100,96 +98,61 @@ func pickAtRandom(r *rand.Rand, names []string, skip string) string {
 	}
 }
 
-var (
-	errDataMismatch        = errors.New("pss: data sent and received are not equal")
-	errWebsocketConnection = errors.New("pss: websocket connection terminated with an error")
-)
-
 var (
 	testData  = []byte("Hello Swarm :)")
 	testTopic = "test"
 )
 
 func (c *Check) testPss(nodeAName, nodeBName string, clients map[string]*bee.Client, o Options) error {
 	ctx, cancel := context.WithTimeout(context.Background(), o.RequestTimeout)
+	defer cancel()
 
 	nodeA := clients[nodeAName]
 	nodeB := clients[nodeBName]
 
 	addrB, err := nodeB.Addresses(ctx)
 	if err != nil {
-		cancel()
 		return err
 	}
 
 	batchID, err := nodeA.GetOrCreateMutableBatch(ctx, o.PostageTTL, o.PostageDepth, o.PostageLabel)
 	if err != nil {
-		cancel()
 		return fmt.Errorf("node %s: batched id %w", nodeAName, err)
 	}
 	c.logger.Infof("node %s: batched id %s", nodeAName, batchID)
 
-	ch, close, err := listenWebsocket(ctx, nodeB.Host(), testTopic, c.logger)
+	ch, closer, err := wslistener.ListenWebSocket(ctx, nodeB, fmt.Sprintf("/pss/subscribe/%s", testTopic), c.logger)
 	if err != nil {
-		cancel()
 		return err
 	}
 
 	c.logger.Infof("pss: sending test data to node %s and listening on node %s", nodeAName, nodeBName)
 
+	defer closer()
+
 	tStart := time.Now()
 	err = nodeA.SendPSSMessage(ctx, addrB.Overlay, addrB.PSSPublicKey, testTopic, o.AddressPrefix, testData, batchID)
 	if err != nil {
-		close()
-		cancel()
 		return err
 	}
+	c.logger.Infof("pss: test data sent successfully to node %s. Waiting for response from node %s", nodeAName, nodeBName)
 
-	msg, ok := <-ch
-	if ok {
-		if msg == string(testData) {
-			c.logger.Info("pss: websocket connection received correct message")
-			c.metrics.SendAndReceiveGauge.WithLabelValues(nodeAName, nodeBName).Set(time.Since(tStart).Seconds())
-		} else {
-			err = errDataMismatch
+	for {
+		select {
+		case <-time.After(1 * time.Minute):
+			return fmt.Errorf("correct message not received after %s", 1*time.Minute)
+		default:
+			msg, ok := <-ch
```

Comment:
Select will never block, and it will never reach `time.After`, because every loop it will fallthrough to default case. Maybe something like this will work better:
```
		select {
		case <-time.After(1 * time.Minute):
			return fmt.Errorf("correct message not received after %s", 1*time.Minute)
		case msg, ok := <-ch:
			if !ok {
				return fmt.Errorf("ws closed before receiving correct message")
			}

			stringData := string(testData)

			if msg != stringData {
				c.logger.Warningf("pss: received incorrect message. trying again. want %s, got %s", stringData, msg)
			}
			c.logger.Info("pss: websocket connection received correct message")
			c.metrics.SendAndReceiveGauge.WithLabelValues(nodeAName, nodeBName).Set(time.Since(tStart).Seconds())
			return nil
		}
		```